src/routes/safe/components/Transactions/helpers/TxParamsState/index.tsx

```diff
@@ -0,0 +1,163 @@
+import { ReactNode, useState } from 'react'
+import styled from 'styled-components'
+import { EstimationStatus, useEstimateTransactionGas } from 'src/logic/hooks/useEstimateTransactionGas'
+import { TxParameters } from 'src/routes/safe/container/hooks/useTransactionParameters'
+import { EditableTxParameters } from 'src/routes/safe/components/Transactions/helpers/EditableTxParameters'
+import { extractSafeAddress } from 'src/routes/routes'
+import { ReviewInfoText } from 'src/components/ReviewInfoText'
+import ExecuteCheckbox from 'src/components/ExecuteCheckbox'
+import { useEstimationStatus } from 'src/logic/hooks/useEstimationStatus'
+import { Errors, logError } from 'src/logic/exceptions/CodedException'
+import { ButtonStatus, Modal } from 'src/components/Modal'
+import { lg, md } from 'src/theme/variables'
+import { TxParametersDetail } from '../TxParametersDetail'
+
+type Props = {
+  children: ReactNode
+  txData: string
+  txValue?: string
+  txType?: string
+  onSubmit: (txParams: TxParameters, delayExecution?: boolean) => void
+  onBack?: (...rest: any) => void
+  submitText?: string
+}
+
+const Container = styled.div`
+  padding: 0 ${lg} ${md};
+`
+
+export const TxParamsState = ({
+  children,
+  txData,
+  txValue = '0',
+  txType,
+  onSubmit,
+  onBack,
+  submitText,
+}: Props): React.ReactElement => {
+  const [manualSafeTxGas, setManualSafeTxGas] = useState('0')
+  const [manualGasPrice, setManualGasPrice] = useState<string | undefined>()
+  const [manualGasLimit, setManualGasLimit] = useState<string | undefined>()
+  const [executionApproved, setExecutionApproved] = useState<boolean>(true)
+  const safeAddress = extractSafeAddress()
+  const isSpendingLimit = txType === 'spendingLimit'
+
+  const {
+    gasCostFormatted,
+    gasPriceFormatted,
+    gasMaxPrioFeeFormatted,
+    gasLimit,
+    gasEstimation,
+    txEstimationExecutionStatus,
+    isExecution,
+    isCreation,
+    isOffChainSignature,
+  } = useEstimateTransactionGas({
+    txData,
+    txRecipient: safeAddress,
+    txType: txType,
+    txAmount: txValue,
+    safeTxGas: manualSafeTxGas,
+    manualGasPrice,
+    manualGasLimit,
+  })
+
+  const [submitStatus, setSubmitStatus] = useEstimationStatus(txEstimationExecutionStatus)
+
+  const doExecute = executionApproved && isExecution
+
+  const onClose = (txParameters: TxParameters) => {
+    const oldGasPrice = gasPriceFormatted
+    const newGasPrice = txParameters.ethGasPrice
+    const oldMaxPrioFee = gasMaxPrioFeeFormatted
+    const newMaxPrioFee = txParameters.ethMaxPrioFee
+    const oldSafeTxGas = gasEstimation
+    const newSafeTxGas = txParameters.safeTxGas
+
+    if (newGasPrice && oldGasPrice !== newGasPrice) {
```

Comment:
In these conditions here, why do you need to check the first var, e.g. why not just `if (oldGasPrice !== newGasPrice) { ... }`?